Implemented mobile responsiveness for Board page (1 column for mobile, 3 for desktop)

cashums opened the pull request:
Cards for Board members appear in a 3-column wide grid on desktop and 1-column wide grid on mobile.
<img width="1843" height="895" alt="image" src="https://github.com/user-attachments/assets/f13adc9f-1602-4a1a-a9c7-98871f3dff28" />
<img width="543" height="867" alt="image" src="https://github.com/user-attachments/assets/44efdc50-13ff-43dd-b46d-9bf2d32a5acc" />

hhannahhwang reviewed (commented):
<img width="866" height="1414" alt="image" src="https://github.com/user-attachments/assets/3688050a-40e1-4573-8b44-9aecaa4c8189" />

This is what most of the mobile screens look like on my end, could we decrease the card sizes a bit for smaller screens. You would be modifying the "Card.tsx" component. 

Could you also make the header responsive -- just adjusting text size should be fine. 

Make sure to PULL before you make edits!

cashums commented:
Addressed comments. Size of Header text as well as Card sizing is now more responsive for mobile screens.

<img width="539" height="550" alt="image" src="https://github.com/user-attachments/assets/c2568fb4-d44c-4e82-aa38-40c9a40e9023" />
<img width="1839" height="818" alt="image" src="https://github.com/user-attachments/assets/2d1b0b0f-98a2-4be1-ac14-ad5afb6a96a9" />
